### PR TITLE
adding intitial github workflow from chatgpt

### PR DESCRIPTION
**Pull request by pauldalyiii**
Please review this initial check in and propose changes

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+
+    - name: Set up JDK 17
+      uses: actions/setup-java@v3
+      with:
+        distribution: 'adopt'
+        java-version: '17'
```
Comment (bloomkd46) [inline .github/workflows/build.yml]:
```suggestion
        java-version: '22'
```

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+
+    - name: Set up JDK 17
```
Comment (bloomkd46) [inline .github/workflows/build.yml]:
```suggestion
    - name: Set up JDK 22
```

**Review by bloomkd46 — CHANGES_REQUESTED** [review]
Update the java version to 22

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
+name: Java CI with Gradle
+
+on:
+  push:
+    branches: [ main ]
```
Comment (bloomkd46) [inline .github/workflows/build.yml]:
```suggestion
  push:
```

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  pull_request:
+    branches: [ main ]
```
Comment (bloomkd46) [inline .github/workflows/build.yml:6]:
```suggestion
  pull_request:
```

**Review by bloomkd46 — CHANGES_REQUESTED** [review]
Modified workflow to run on all branches

File: .github/workflows/build.yml (hunk 1 above)
Comment (pauldalyiii) [reply .github/workflows/build.yml]: addressed

File: .github/workflows/build.yml (hunk 2 above)
Comment (pauldalyiii) [reply .github/workflows/build.yml]: addressed

**Pull request merged**